modules/host/storageobligations.go

```diff
@@ -203,18 +298,23 @@ func (h *Host) addStorageObligation(so *storageObligation) error {
 	if !exists {
 		h.log.Critical("addStorageObligation called with an obligation that is not locked")
 	}
-	// Expensive Sanity check - obligation being added should have valid
-	// tranasctions.
-	if build.DEBUG {
-		for _, txn := range so.OriginTransactionSet {
-			err := txn.StandaloneValid(h.blockHeight)
-			if err != nil {
-				h.log.Critical("invalid transaction is being added in a storage obligation")
-			}
-		}
+	// Sanity check - There needs to be enough time left on the file contract
+	// for the host to safely submit the file contract revision.
+	if so.expiration()-resubmissionTimeout <= h.blockHeight {
```

Comment:
A closer look revealed that the logic was incorrect, and also that there were ways to construct the conditional such that only addition was used.

thx